qiskit/algorithms/optimizers/steppable_optimizer.py

```diff
@@ -0,0 +1,252 @@
+from abc import abstractmethod, ABC
+from dataclasses import dataclass, field
+from typing import Dict, Any, Union, Callable, Optional, Tuple, List
+
+from .optimizer import Optimizer, POINT, OptimizerResult
+import numpy as np
+
+CALLBACK = Callable[[int, np.ndarray, float, float], None]
```

Comment:
I believe the callback differs for the optimizers (see e.g. gradient descent and SPSA), probably it makes more sense to just keep the definitions inside the files of the optimizers.

Reply:
Do I typehint it to just be a callable?